lmdeploy/pytorch/models/deepseek_vl2.py

```diff
@@ -0,0 +1,560 @@
+
+        return dict(
+            input_ids=input_ids,  # [b, T]
+            position_ids=position_ids,
+            past_key_values=past_key_values,
+            attn_metadata=attn_metadata,
+            pixel_values=pixel_values,  # [b, max_n_images, 3, height, width]
+            images_spatial_crop=images_spatial_crop,  # [b, max_n_images, 2]
+            image_mask=images_seq_mask,  # [b, T]
+            inputs_embeds=inputs_embeds,
+        )
+
+    def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
+        """load weights."""
+
+        lang_prefix = 'language.'
+        lang_prefix_length = len(lang_prefix)
+        new_weights = dict()
+        params_dict = dict(self.named_parameters())
+
+        for name, loaded_weight in weights:
+            if name.startswith(lang_prefix):
+                new_key = name[lang_prefix_length:]
+                new_weights[new_key] = loaded_weight
+                continue
+
+            if 'qkv' in name and 'vision' not in name:
+                param = params_dict[name]
+                q, k, v = param.weight_spliter(loaded_weight)
+                load_weight(param, q, shard_id='q')
+                load_weight(param, k, shard_id='k')
+                load_weight(param, v, shard_id='v')
+            else:
+                param = params_dict[name]
+                load_weight(param, loaded_weight)
+
+        self.language.load_weights(new_weights.items())
+
+    def get_input_processor(self) -> BaseModelInputProcessor:
+        """get input processor."""
+        return self.input_processor
+
+
+class DeepSeekVLV2InputProcessor(BaseModelInputProcessor):
+    """deepseek-vl2 input processor."""
+
+    def __init__(self, config: PretrainedConfig, dtype) -> None:
+        self.config = config
+        self.dtype = dtype
+        vision_config = config.vision_config
+        self.patch_size = vision_config.patch_size
+
+    def preprocess_input(self,
+                         input_ids: List[int],
+                         input_multimodals: List[Dict[str, Any]] = None,
+                         **kwargs) -> PreprocessInputResult:
+        """prepare multimodal input."""
+        if input_multimodals is None or len(input_multimodals) == 0:
+            return input_ids, input_multimodals
+
+        input_imgs = []
+        for input_mm in input_multimodals:
+            pixel_values = input_mm['pixel_values'].to(self.dtype)
+            offset = input_mm['offset']
+            image_token_id = input_mm.get('image_token_id', 0)
+            num_pad = input_mm['image_tokens']
+            images_spatial_crop = input_mm.get('images_spatial_crop', None)
+            images_seq_mask = input_mm.get('images_seq_mask', None)
+            if isinstance(num_pad, torch.Tensor):
+                num_pad = num_pad.item()
+
+            # in multi-round conversations, lmdeploy won't add bos_id for non-first-round messages
```

Comment:
then we may need to construct `images_seq_mask` in here instead of using `images_seq_mask` from upstream preprocessor.